### DISCUSSION TITLE
How to get netmask?

### DISCUSSION THREAD

**Discussion by ViniScardoso** [oshi/oshi#2263] [Q&A] [answered]
Hi, I'm trying to get the ipv4 and ipv6 netmasks but the only method that I found was "getSubnetMasks" but it returns a Short value. Does anyone know how to properly get this informations?

**Comment by dbwiddis** [comment, marked as answer]
Hey @ViniScardoso ... the netmasks are between 0 and 32, as you would see in CIDR notation.

So a value of 32 corresponds to a mask of 255.255.255.255.  A value of 31 would be 255.255.255.254.  A value of 24 would be 255.255.255.0.  And so on.

The IPv6 prefix lengths are treated similarly, but are between 0 and 128.

These fields are pulled directly from the interfaces iterated in Java's `NetworkInterface` class [`getInterfaceAddresses()`](https://docs.oracle.com/javase/8/docs/api/java/net/NetworkInterface.html#getInterfaceAddresses--) method.  The values come from [`getNetworkPrefixLength()`](https://docs.oracle.com/javase/8/docs/api/java/net/InterfaceAddress.html#getNetworkPrefixLength--).
